### PR TITLE
[tests] split python e2e tests into chunks

### PR DESCRIPTION
**Pull request by EndangeredMassa**
Split python tests into two chunks. They take <10m each.

### CODE REVIEW

File: packages/python/test/integration-1.test.ts [hunk 1]
```diff
@@ -71,30 +71,11 @@ it('should match the probes against Python dev servers', async () => {
...
 
-const testsThatFailToBuild = new Map([
-  ['30-fail-build-invalid-pipfile', 'Unable to parse Pipfile.lock'],
-  [
-    '31-fail-build-invalid-python36',
-    'Python version "3.6" detected in Pipfile.lock is discontinued and must be upgraded.',
-  ],
-]);
+const allFixtures = fs.readdirSync(fixturesPath);
+const chunkFixtures = allFixtures.slice(0, 10);
```
Comment (dan-stowell) [inline packages/python/test/integration-1.test.ts]: Why slice the first 10 fixtures?

Comment (dan-stowell) [reply packages/python/test/integration-1.test.ts]: Oh, I see! `integration-2.test.ts` runs the rest.

**Review by dan-stowell — DISMISSED** [review]
Approving because faster tests now sounds good!

Having to write code to read the filesystem and slice the tests into chunks seems error-prone over time (what if we write new tests and those new tests push the slow tests all into one chunk?). But I assume this is the most straightforward way to split things up!

File: packages/python/test/integration-1.test.ts (hunk 1 above)
Comment (EndangeredMassa) [reply packages/python/test/integration-1.test.ts]: Yeah, this is not the best approach

**Review by styfle — CHANGES_REQUESTED** [review]
Lets follow the pattern from `@vercel/static-build` to keep it DRY

https://github.com/vercel/vercel/blob/5b88f673f89c44b9e91e44247e29be6bed43805a/packages/static-build/test/integration-1.test.js#L1

**Comment by dan-stowell** [comment]
@styfle, looks like @EndangeredMassa put the test-splitting logic into a single module, this look good to land?

**Review by dan-stowell — DISMISSED** [review]
👍 ! Again!

**Comment by styfle** [comment]
@dan-stowell Lets verify all the expected test fixtures actually run

File: packages/python/test/integration-1.test.ts [hunk 2]
```diff
@@ -1,15 +1,14 @@
-const fs = require('fs');
-const path = require('path');
+require('./integration-setup')(1);
+
+// NOTE: The fixture `00-requrest-path` has special case handling, which is why
```
Comment (styfle) [inline packages/python/test/integration-1.test.ts]:
```suggestion
// NOTE: The fixture `00-request-path` has special case handling, which is why
```

Took me a second to realize why I couldn't find it 😆

File: packages/python/test/integration-setup.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  let chunkedFixtures = allFixtures;
+  if (typeof groupIndex !== 'undefined') {
+    chunkedFixtures = intoChunks(1, 2, allFixtures)[groupIndex - 1];
+
+    console.log('testing group', groupIndex, chunkedFixtures);
+  }
+
+  // filter out '00-request-path' because it has special handling in "integration-1.test.ts"
+  // we wait until after chunking to filter it out so that chunk 1 will have one less test in the chunk
+  // which is backfilled by the special handling
```
Comment (styfle) [inline packages/python/test/integration-setup.js]: I actually don't think it needs to be filtered out. I think one is testing dev and one is testing deploy

Comment (EndangeredMassa) [reply packages/python/test/integration-setup.js]: Good point. Updated: [`2057ab7` (#9771)](https://github.com/vercel/vercel/pull/9771/commits/2057ab7626539ba36faa369ea29897eaab8ec942) + [`caf648d` (#9771)](https://github.com/vercel/vercel/pull/9771/commits/caf648d71db1a1ffde1e491420d8b436adf57d6b)

**Comment by EndangeredMassa** [comment]
I checked the tests that actually ran vs. main. They all appear to be accounted for.

# main

```
✓ should match the probes against Python dev servers (9781 ms)
✓ should build 00-request-path (88715 ms)
✓ should build 01-cowsay (44408 ms)
✓ should build 02-wsgi-pipenv (57543 ms)
✓ should build 03-env-vars (66271 ms)
✓ should build 04-cowsay-pipfile (41933 ms)
✓ should build 05-cowsay-pipfile-local (53996 ms)
✓ should build 06-content-type (39552 ms)
✓ should build 08-django (54913 ms)
✓ should build 09-url-params (47509 ms)
✓ should build 11-asgi (51394 ms)
✓ should build 12-asgi-function (41037 ms)
✓ should build 14-unicode-handler (35716 ms)
✓ should build 15-binary-handler (50692 ms)
✓ should build 20-multivalue-header (54086 ms)
✓ should build 25-exclude-files (54856 ms)
✓ should build 27-square-brackets (58055 ms)
✓ should build 28-pydantic (54676 ms)
✓ should build 29-flask2 (43935 ms)
✓ should fail to build 30-fail-build-invalid-pipfile (57868 ms)
✓ should fail to build 31-fail-build-invalid-python36 (38014 ms)
✓ should build 32-fail-duplicate-header (39515 ms)
```

# PR Part 1

```
✓ should build 00-request-path (64577 ms)
✓ should build 01-cowsay (49501 ms)
✓ should build 02-wsgi-pipenv (48520 ms)
✓ should build 03-env-vars (46116 ms)
✓ should build 04-cowsay-pipfile (55972 ms)
✓ should build 05-cowsay-pipfile-local (44727 ms)
✓ should build 06-content-type (43278 ms)
✓ should build 08-django (50645 ms)
✓ should build 09-url-params (53153 ms)
✓ should build 11-asgi (50234 ms)
✓ should build 12-asgi-function (43192 ms)
✓ should match the probes against Python dev servers (10479 ms)
```

# PR Part 2

```
✓ should build 14-unicode-handler (40325 ms)
✓ should build 15-binary-handler (41139 ms)
✓ should build 20-multivalue-header (71425 ms)
✓ should build 25-exclude-files (43946 ms)
✓ should build 27-square-brackets (39593 ms)
✓ should build 28-pydantic (55974 ms)
✓ should build 29-flask2 (52450 ms)
✓ should fail to build 30-fail-build-invalid-pipfile (36722 ms)
✓ should fail to build 31-fail-build-invalid-python36 (38146 ms)
✓ should build 32-fail-duplicate-header (53318 ms)
```

**Review by styfle — APPROVED** [review]
Thanks!

**Review by cb1kenobi — APPROVED** [review]

**Pull request merged**